pdns/syncres.cc

```diff
@@ -283,6 +283,9 @@ bool SyncRes::doOOBResolve(const DNSName &qname, const QType &qtype, vector<DNSR
 void SyncRes::doEDNSDumpAndClose(int fd)
 {
   FILE* fp=fdopen(fd, "w");
+  if (!fp) {
+    return;
```

Comment:
You are right, but the error should get to the right channel, so we would need to return a error to the caller or move the `fdopen()` to the calling function. Similar functions (`MemRecursorCache::doDumpNSSpeeds()`, `MemRecursorCache::doDump()`, `RecursorPacketCache::doDump()`, ...) don't seem to care.